src/04_output_TMF.R

```diff
@@ -0,0 +1,198 @@
+View(TMF_dict_detail)
+
+
+# TMF_dict_detail %>% pull(`Field Name`)
+# 
+# # Your vector of strings
+# strings <- TMF_dict_detail %>% pull(`Field Name`)
+# 
+# # Create the mapping
+# mapping <- setNames(strings, strings)
+# 
+# # Custom print function
+# formatted_output <- paste0('"', names(mapping), '" = "', mapping, '"', collapse = ", ")
+# 
+# # Print the formatted string
+# cat(formatted_output)
+
+# then I can copy the results to our new switch function
+# maybe better just do it in CSV file.
+# or I should use those as labels in create_dictionary function. liek c(POSTAL_CODE = "Postal code")
+# finally use chatgpt to translate the switch function to a case_when
+create_item = function(x) {
+  # Using case_when to replace the switch statement
+  y <- case_when(
+    x == "Postal code" ~ "POSTALCODE",
+    x == "Birth Date" ~ "BIRTH_DATE",
+    x == "Retired Date" ~ "RET_DATE",
+    x == "Latitude" ~ "LATITUDE",
+    x == "Longitude" ~ "LONGITUDE",
+    x == "Census Division" ~ "CD",
+    x == "Census Subdivision" ~ "CSD",
+    x == "Census Subdivision Name" ~ "MUN_NAME",
+    x == "Census Metropolitan Area or Census Agglomeration Area" ~ "CMACA",
+    x == "Dissemination Area (DA)" ~ "DA",
+    x == "Census Tract (CT)" ~ "CT",
+    x == "Dissemination Block (DB)" ~ "DB",
+    x == "Designated Place Name (DPL)" ~ "DPL",
+    x == "Population Centre (POPCTR)" ~ "POPCTR",
+    x == "Development Region (DR)" ~ "DR",
+    x == "Health Authority (HA)" ~ "HA",
+    x == "Health Service Delivery Area (HSDA)" ~ "HSDA",
+    x == "Local Health Area (LHA)" ~ "LHA",
+    x == "Community Health Service Area (CHSA)" ~ "CHSA",
+    x == "Pre-2018 Local Health Area" ~ "LHA_PRE_2018",
+    x == "Micro Health Area" ~ "MHA",
+    x == "Ministry of Children & Families Region (MCFD)" ~ "MCFD",
+    x == "Ministry of Children & Families Service Delivery Area (MCFD_SDA)" ~ "MCFD_SDA",
+    x == "Ministry of Children & Families Local Service Area (MCFD_LSA)" ~ "MCFD_LSA",
+    x == "College Region" ~ "CR",
+    x == "School District (SD)" ~ "SD",
+    x == "School District Trustee Electoral Area (TEA)" ~ "TEA",
+    x == "Provincial Electoral District (PED)" ~ "PED",
+    x == "Federal Electoral District (FED)" ~ "FED",
+    x == "Police Services Respondent Area (RESP)" ~ "RESP",
+    x == "Tourism Region" ~ "TOURISM",
+    x == "Games Zone" ~ "GZ",
+    x == "Community Name" ~ "COMM_NAME",
+    x == "Modified Census Subdivision Name" ~ "Modified_MUN_NAME",
+    x == "Modified Full Census Subdivision" ~ "CDCSD",
+    TRUE ~ NA_character_  # Default case if no match
+  )
+  
+  return(y)
+}
+
+TMF_dict_detail <- TMF_dict_detail %>%
+  mutate(item_short = create_item(`Field Name`))
+  
+# for the datadictionary created from datadictionary function, we also need to create a shorten item name since some items have year as sufix such as CD_2021. 
+TMF_dict <- TMF_dict %>% 
+  mutate( item_short = str_remove(item, "_\\d{4}")) 
```

Comment:
tiny change but maybe

```
# for the datadictionary created from datadictionary function, we also need to create a shorten item name since some items have year as suffix such as CD_2021. 
TMF_dict <- TMF_dict %>% 
  mutate( item_short = str_remove(item, "_\\d{4}$")) 
```

Reply:
Great. I added "$" to ensure that only the digits at the strings' end are removed.